exec/java-exec/src/main/java/org/apache/drill/exec/store/easy/text/writer/TextRecordWriter.java

```diff
@@ -93,29 +112,35 @@ public void startNewSchema(List<String> columnNames) throws IOException {
       // we need to re-apply file permission
       DataOutputStream fos = fs.create(fileName);
       storageStrategy.applyToFile(fs, fileName);
+      logger.debug("Created file: {}.", fileName);
+
+      this.writer = new CsvWriter(fos, writerSettings);
 
-      stream = new PrintStream(fos);
-      logger.debug("Created file: {}", fileName);
-    } catch (IOException ex) {
-      logger.error("Unable to create file: " + fileName, ex);
-      throw ex;
+    } catch (IOException e) {
+      throw new IOException(String.format("Unable to create file: %s.", fileName), e);
     }
-    index++;
 
-    stream.println(Joiner.on(fieldDelimiter).join(columnNames));
+    fileNumberIndex++;
+
+    if (writerSettings.isHeaderWritingEnabled()) {
+      writer.writeHeaders(
+        IntStream.range(0, schema.getFieldCount())
```

Comment:
Nit: it can be a little bit optimized to avoid looking up elements from the list by index:
```suggestion
        StreamSupport.stream(schema.spliterator(), false)
```

Reply:
Good point, thanks for noticing. Fixed.